internal/flags/helpers.go

```diff
@@ -263,3 +265,37 @@ func AutoEnvVars(flags []cli.Flag, prefix string) {
 		}
 	}
 }
+
+// CheckEnvVars iterates over all the environment variables and checks if any of
+// them look like a CLI flag but is not consumed. This can be used to detect old
+// or mistyped names.
+func CheckEnvVars(ctx *cli.Context, flags []cli.Flag, prefix string) {
+	known := make(map[string]string)
+	for _, flag := range flags {
+		docflag, ok := flag.(cli.DocGenerationFlag)
+		if !ok {
+			continue
+		}
+		for _, envvar := range docflag.GetEnvVars() {
+			known[envvar] = flag.Names()[0]
+		}
+	}
+	keyvals := os.Environ()
+	sort.Strings(keyvals)
+
+	for _, keyval := range keyvals {
+		key := strings.Split(keyval, "=")[0]
+		if !strings.HasPrefix(key, prefix) {
+			continue
+		}
+		if flag, ok := known[key]; ok {
+			if ctx.Count(flag) > 0 {
+				log.Info("Config environment variable found", "envvar", key, "shadowedby", "--"+flag)
+			} else {
+				log.Info("Config environment variable found", "envvar", key)
```

Comment:
In some cases. cmd line params are sensitive (password for remote influxdb databases, password for `--unlock`), but this will echo them in logs, right? 

Might not be the best idea -- so far, we've been kind rightfully lax about asking people for their logs, because we know they contain nothing interesting. 

Ah wait, it's only the `key`, not the value? If so it's all good!

Reply:
Yes, only key for the exact reason.